File: python/set07/set07.py
def extract_words(text):
    return text.split()

# Problem 1: Testing
text = "This is a sample text with several words."
print(extract_words(text))

######################################################################################
def csv_to_array(csv_text):
    # Split the text into lines
    lines = csv_text.strip().split("\n")

    # Split each line into values
    array = [line.split(",") for line in lines]
    
    return array

# Problem 2: Testing
csv_text = "name,age,city\nJohn,30,New York\nJane,25,Los Angeles"
print(csv_to_array(csv_text))

######################################################################################

def string_to_array(s):
    return list(s)

# Problem 3: Testing
text = "hello"
print(string_to_array(text))  # Output: ['h', 'e', 'l', 'l', 'o']

######################################################################################
# Create a function that will convert a string in an array containing the ASCII codes of each character

def convertingStringToASCII(words):
    words = str.split(" ")
    for word in words:
        first_letter = word[0]
        if 'A' <= first_letter <= 'Z':
            first_letter = chr(ord(first_letter) + 32)
        elif 'a' <= first_letter <= 'z':
            first_letter = chr(ord(first_letter) - 32)
        
        print(first_letter + word[1:], end=' ')

# Problem 4: Testing
s = "This is a test of string manipulation"
print(convertingStringToASCII(s))

######################################################################################
# Create a function that will convert an array containing ASCII codes in a string

######################################################################################
# Implement the Caesar cypher

######################################################################################
# Implement the bubble sort algorithm for an array of numbers

File: javascript/set07/set07.js
function extractWords(text) {
    return text.split(/\s+/).filter(Boolean);
}

// Problem 1: Testing
const text = "This is a sample text with several words.";
console.log(extractWords(text));

//////////////////////////////////////////////////////////////////////////////////////////
function csvToArray(csvText) {
    // Split the text into lines
    const lines = csvText.trim().split("\n");

    // Split each line into values
    const array = lines.map(line => line.split(","));
    
    return array;
}

// Problem 2: Testing
const csvText = "name,age,city\nJohn,30,New York\nJane,25,Los Angeles";
console.log(csvToArray(csvText));

//////////////////////////////////////////////////////////////////////////////////////////
function stringToArray(str) {
    return str.split('');
}

// Problem 3: Testing
const sampleText = "hello";
console.log(stringToArray(sampleText)); // Output: ['h', 'e', 'l', 'l', 'o']

//////////////////////////////////////////////////////////////////////////////////////////

function convertingStringToASCII(words) {
    let words = s.split(" ");
    words.forEach((word) => {
        let firstLetter = word.charAt(0);
        if (firstLetter >= 'A' && firstLetter <= 'Z') {
            firstLetter = String.fromCharCode(firstLetter.charCodeAt(0) + 32);
        } else if (firstLetter >= 'a' && firstLetter <= 'z') {
            firstLetter = String.fromCharCode(firstLetter.charCodeAt(0) - 32);
        }
        process.stdout.write(firstLetter + word.substring(1) + " ");
    });
}

// Problem 4: Testing
let s = "This is a test of string manipulation";
console.log(convertingStringToASCII(s))

//////////////////////////////////////////////////////////////////////////////////////////

// Create a function that will convert an array containing ASCII codes in a string

//////////////////////////////////////////////////////////////////////////////////////////
// Implement the Caesar cypher

//////////////////////////////////////////////////////////////////////////////////////////
// Implement the bubble sort algorithm for an array of numbers